### DISCUSSION TITLE
Dereferncing symbolic links in filebrowser

### DISCUSSION THREAD

**Discussion by avidseeker** [davatorium/rofi#1796] [Q&A] [answered]
Is that possible?

**Comment by DaveDavenport** [comment, marked as answer]
A symlink will be prepended with an @ before it name and you should be able to follow it. 

What does not work? 

(I checked a link to  directory and this works and so does link to a file.)

**Comment by avidseeker** [comment]
What I mean by dereferencing is reading the destination of symlink and moving to it. If a have `~/symlink-dir/` linking to `/mnt/` then I want the path to be updated to `/mnt` instead of sticking with `~/symlink-dir`. i.e: I want `cd $(readlink ~/symlink-dir/)` instead of `cd ~/symlink-dir`.

**Reply by DaveDavenport** [reply]
This is not what f.e. nautilus does with symlinks or bash on my shell if you normally enter the director.  ( it also breaks your 'up' flow , e.g. going to '..').
I am not sure if breaking from what seems to be default is smart.

**Comment by avidseeker** [comment]
What I want is something similar to Windows `.lnk` shortcuts. As you said, symbolic links might not the best implementation for this.

Probably `.desktop` entries are the best way to do this. But rofi is launching them with `xdg-open`, and `xdg-open` is bugged (opening them in text editor rather than launching them).

Is there a way launch `.desktop` entries in other app?

**Reply by DaveDavenport** [reply]
Not familiar with this, but I think you can query what is used with xdg-mime:
```bash
xdg-mime query default application/x-desktop
```
and also use xdg-mime to set an application to use for launching it.

**Reply by avidseeker** [reply]
Marking this as answered as of now, since the issue isn't related to Rofi.

Thank you for your support
